### ISSUE TITLE
Normalizing only on the training data

### ISSUE THREAD

**Issue by shayko18** [udacity/sdc-issue-reports#749]
Hi,
I think you have a small (but important) mistake in the code you put in lesson 34 "Search and Classify" in project 5 (CarND-Vehicle-Detection). 
You have calculated the "X_scaler" using all the data and not just with the training set data (Line 142 in search_classify.py)
It will contaminate the test set error estimation.
Thanks

**Comment by baumanab** [comment]
related to #490

**Comment by baumanab** [comment]
Thank you for reporting this.  The code has been updated to avoid data snooping.
